packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts

```diff
@@ -515,159 +530,133 @@ export default class TableResize implements EditorPlugin {
-                for (let i = 0; i < vtable.cells.length; i++) {
-                    for (let j = 0; j < vtable.cells[i].length; j++) {
-                        const cell = vtable.cells[i][j];
-                        if (cell.td) {
-                            if (shouldResizeX) {
-                                const originalWidth: number = cell.td.style.width
-                                    ? parseFloat(
-                                          cell.td.style.width.substr(
-                                              0,
-                                              cell.td.style.width.length - 2
-                                          )
-                                      )
-                                    : cell.td.getBoundingClientRect().right -
-                                      cell.td.getBoundingClientRect().left;
-                                const newWidth = originalWidth * ratioX;
-                                cell.td.style.boxSizing = 'border-box';
-                                if (newWidth >= MIN_CELL_WIDTH) {
-                                    cell.td.style.wordBreak = 'break-word';
-                                    cell.td.style.width = `${newWidth}px`;
-                                }
+    private resizeTable = (mouseX: number, mouseY: number) => {
+        const rect = normalizeRect(this.resizingVtable.table.getBoundingClientRect());
+        const ratioX =
+            1.0 +
+            (this.isRTL
+                ? (this.currentTableVerticalBorder - mouseX) /
+                  (rect.right - this.currentTableVerticalBorder)
+                : (mouseX - this.currentTableVerticalBorder) /
+                  (this.currentTableVerticalBorder - rect.left));
+        const ratioY =
+            1.0 +
+            (mouseY - this.currentTableHorizontalBorder) /
+                (this.currentTableHorizontalBorder - rect.top);
+
+        const shouldResizeX = Math.abs(ratioX - 1.0) > 1e-3;
+        const shouldResizeY = Math.abs(ratioY - 1.0) > 1e-3;
+        if (shouldResizeX || shouldResizeY) {
+            for (let i = 0; i < this.resizingVtable.cells.length; i++) {
+                for (let j = 0; j < this.resizingVtable.cells[i].length; j++) {
+                    const cell = this.resizingVtable.cells[i][j];
+                    if (cell.td) {
+                        if (shouldResizeX) {
+                            // the width of some external table is fixed, we need to make it resizeable
+                            this.resizingVtable.table.style.width = null;
+                            const newWidth = cell.width * ratioX;
+                            cell.td.style.boxSizing = 'border-box';
+                            if (newWidth >= MIN_CELL_WIDTH) {
+                                cell.td.style.wordBreak = 'break-word';
+                                cell.td.style.whiteSpace = 'normal';
+                                cell.td.style.width = `${newWidth}px`;
```

Comment:
Shouldn't this be done when normalized the table?

Reply:
cell.td.style.width = `${newWidth}px` here is to set new width for each cell when we resize the whole table based on the resizing ratio, why should we do it when normalizing the table in the beginning?

Reply:
ok